packages/viewer/src/components/viewer/viewer.tsx

```diff
@@ -310,8 +311,21 @@ export class Viewer {
     await this.unload();
 
     if (this.commands != null && this.dimensions != null) {
-      this.resource = LoadableResource.fromUrn(urn);
-      await this.connectStreamingClient(this.resource);
+      const loadableResource = LoadableResource.fromUrn(urn);
+
+      const isCurrentlyLoadingResource =
+        this.connectingPromise != null &&
+        this.resource != null &&
+        this.resource.id === loadableResource.id;
+      if (isCurrentlyLoadingResource) {
+        return;
+      }
+      this.resource = loadableResource;
+
+      this.connectingPromise = this.connectStreamingClient(this.resource);
+
+      await this.connectingPromise;
+      this.connectingPromise = undefined;
```

Comment:
This should probably be cleared when `unload` is called as well. Otherwise a hanging connection is going to prevent a second call to load without anyway to get the viewer out of that state.